src/DotNet/CustomAttributeReader.cs

```diff
@@ -208,6 +208,8 @@ public struct CustomAttributeReader {
 		/// <returns>A list of <see cref="CANamedArgument"/>s or <c>null</c> if some error
 		/// occurred.</returns>
 		internal static List<CANamedArgument> ReadNamedArguments(ModuleDef module, ref DataReader reader, int numNamedArgs, GenericParamContext gpContext) {
+			if (numNamedArgs > 0x10000)
```

Comment:
The check should be in `ReadNamedArguments()`.

It should not limit it to 64K args. You should instead check that there's enough bytes left to read all named args. Each named arg is at least 4 bytes in size so make sure count is not negative and also < 0x4000_0000 and also that there's at least count*4 bytes left in the reader: `(uint)count < 0x4000_0000 && count * 4 < bytes left`